llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5256,6 +5309,13 @@ LoopVectorizationCostModel::selectInterleaveCount(ElementCount VF,
   if (!isScalarEpilogueAllowed())
     return 1;
 
+  // Do not interleave if EVL is preferred and no User IC is specified.
+  if (useVPIWithVPEVLVectorization()) {
```

Comment:
nit: more explicit to check tail-folding style directly here or rename helper to something like `foldTailWithEVL`?